src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -573,11 +574,95 @@ class GitVersionPluginTests extends Specification {
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+            subprojects {
+                apply plugin: 'com.palantir.git-version'
+                version gitVersion()
+            }
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+        Git git = Git.init().setDirectory(projectDir).call()
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        git.tag().setAnnotated(false).setName('1.0.0').call()
+        git.tag().setAnnotated(true).setName('2.0.0').call()
+        git.tag().setAnnotated(false).setName('3.0.0').call()
+
+        when:
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output.contains(":printVersion\n2.0.0\n")
+    }
+
+    def 'test multiple tags on same commit - most recent annotated tag' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+            subprojects {
+                apply plugin: 'com.palantir.git-version'
+                version gitVersion()
+            }
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+        Git git = Git.init().setDirectory(projectDir).call()
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        PersonIdent ident = new PersonIdent("name", "email@example.com")
+        git.tag().setAnnotated(true).setTagger(new PersonIdent(ident, new Date() - 2)).setName('1.0.0').call()
+        git.tag().setAnnotated(true).setTagger(new PersonIdent(ident, new Date())).setName('2.0.0').call()
+        git.tag().setAnnotated(true).setTagger(new PersonIdent(ident, new Date() - 1)).setName('3.0.0').call()
+
+        when:
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output.contains(":printVersion\n2.0.0\n")
+    }
+
+    def 'test multiple tags on same commit - smaller tag is chosen' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            version gitVersion()
+            subprojects {
+                apply plugin: 'com.palantir.git-version'
+                version gitVersion()
+            }
+        '''.stripIndent()
+        gitIgnoreFile << 'build'
+        Git git = Git.init().setDirectory(projectDir).call()
+        git.add().addFilepattern('.').call()
+        git.commit().setMessage('initial commit').call()
+        git.tag().setAnnotated(false).setName('2.0.0').call()
+        git.tag().setAnnotated(false).setName('1.0.0').call()
+        git.tag().setAnnotated(false).setName('3.0.0').call()
+
+        when:
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output.contains(":printVersion\n1.0.0\n")
+    }
+
```

Comment:
Could we add a test where the git tree contains a symlink to another file? Apparently that was historically an incompatibility between real git and jgit

Reply:
turns out there is already a test for that https://github.com/palantir/gradle-git-version/blob/develop/src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy#L277